src/read.c

```diff
@@ -4268,24 +4272,29 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
     if (alphaItemCount != colorItemCount) {
         // Not all the color items had an alpha auxiliary attached to it. Report this case as an image without alpha channel.
         avifFree(alphaItemIndices);
-        *alphaItem = NULL;
+        *alphaItemIndex = -1;
         *isAlphaItemInInput = AVIF_FALSE;
         return AVIF_RESULT_OK;
     }
-    const avifResult result = avifMetaFindOrCreateItem(meta, maxItemID + 1, alphaItem); // Create new empty item.
+    avifDecoderItem * alphaItem;
+    const avifResult result = avifMetaFindOrCreateItem(meta, maxItemID + 1, &alphaItem); // Create new empty item.
     if (result != AVIF_RESULT_OK) {
         avifFree(alphaItemIndices);
         *isAlphaItemInInput = AVIF_FALSE;
         return result;
     }
-    memcpy((*alphaItem)->type, "grid", 4); // Make it a grid and register alpha items as its tiles.
-    (*alphaItem)->width = colorItem->width;
-    (*alphaItem)->height = colorItem->height;
+    // avifMetaFindOrCreateItem() could invalidate all existing item pointers. So repopulate the colorItem pointer.
+    colorItem = &meta->items.item[colorItemIndex];
+
+    memcpy(alphaItem->type, "grid", 4); // Make it a grid and register alpha items as its tiles.
+    alphaItem->width = colorItem->width;
+    alphaItem->height = colorItem->height;
     for (uint32_t i = 0; i < alphaItemCount; ++i) {
         avifDecoderItem * item = &meta->items.item[alphaItemIndices[i]];
-        item->dimgForID = (*alphaItem)->id;
+        item->dimgForID = alphaItem->id;
     }
     avifFree(alphaItemIndices);
+    *alphaItemIndex = meta->items.count - 1;
```

Comment:
Nit/optional: It seems clearer to do this assignment immediately after the `avifMetaFindOrCreateItem()` call. We can probably wait until we know the call is successful, after line 4285.

Reply:
Done.